src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -855,6 +872,76 @@ func validateListener(
 				external.Subdomain,
 				"TLS requires specifying a subdomain"))
 	}
+	if tlsEnabled && clientCACertRef != nil {
+		if !requireClientAuth {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("requireClientAuth"),
+					requireClientAuth,
+					"Enabled has to be set to true for RequireClientAuth if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Name == "" {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Name must be provided if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Kind != "" && !strings.EqualFold(clientCACertRef.Kind, "Secret") {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Kind must be set to secret if set in ClientCACertRef"))
+		}
+
+		if len(allErrs) == 0 {
+			if clientCACertRef.Namespace == "" {
+				clientCACertRef.Namespace = clusterNamespace
+			}
+
+			secret := &corev1.Secret{}
+			err := kclient.Get(context.TODO(), types.NamespacedName{Name: clientCACertRef.Name, Namespace: clientCACertRef.Namespace}, secret)
+			if err != nil {
+				allErrs = append(allErrs,
+					field.Invalid(
+						path.Child("clientCACertRef"),
+						clientCACertRef,
+						"Failed to get secret: "+err.Error()))
+				return allErrs
+			}
+
+			crt, found := secret.Data[cmmeta.TLSCAKey]
+			if !found {
+				allErrs = append(allErrs,
+					field.Invalid(
+						path.Child("clientCACertRef"),
+						clientCACertRef,
+						"ca.crt must be set in the client CA secret"))
+				return allErrs
+			}
+
+			pb, _ := pem.Decode(crt)
```

Comment:
Does Redpanda only support pem?

Reply:
No requirement on support other, e.g. DER. I will check.

Reply:
Only PEM for now. See https://redpandadata.slack.com/archives/C04LULYFG1X/p1681136849120679